e3db/tsv1_auth.py

```diff
@@ -67,18 +65,24 @@ def create_tsv1_signature(self, r: requests.Request, nonce: UUID, timestamp: int
         """
 
         # Generate header values
-        header_string = f"{self.AUTHENTICATION_METHOD}; {self.public_b64}; {timestamp}; {nonce}; uid:{self.client_id}"
+        header_string = f"{self.AUTHENTICATION_METHOD}; {public_b64}; {timestamp}; {nonce}; uid:{client_id}"
+        
+        # Parse and sort query parameters
+        url_components = urlparse(r.url)
+        query_components = parse_qsl(url_components.query)
+        query_components.sort()
+        query_string = urlencode(query_components)
+
+        call_path = url_components.path
 
-        call_path = r.url
-        query_string = r.params
         call_method = r.method
 
         # Hash header values
         string_to_hash = f"{call_path}; {query_string}; {call_method}; {header_string}"
 
         # Sign hash
         string_to_sign = BaseCrypto.hashString(string_to_hash)
-        full_signature = SodiumCrypto.sign_string(string_to_sign, self.private_b64_decoded)
+        full_signature = SodiumCrypto.sign_string(string_to_sign, private_b64_decoded)
```

Comment:
Reminder to update the notes design doc with hard won information you've learned as part of figuring out these lower level details of how to sort query params for tsv1 signing